extensions/leveldb/deployment/src/main/java/org/apache/camel/quarkus/component/leveldb/deployment/LeveldbProcessor.java

```diff
@@ -16,31 +16,41 @@
  */
 package org.apache.camel.quarkus.component.leveldb.deployment;
 
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
-import io.quarkus.deployment.annotations.ExecutionTime;
-import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.pkg.steps.NativeBuild;
-import org.apache.camel.quarkus.core.JvmOnlyRecorder;
-import org.jboss.logging.Logger;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import org.apache.camel.support.DefaultExchangeHolder;
 
 class LeveldbProcessor {
 
-    private static final Logger LOG = Logger.getLogger(LeveldbProcessor.class);
     private static final String FEATURE = "camel-leveldb";
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
-    /**
-     * Remove this once this extension starts supporting the native mode.
-     */
-    @BuildStep(onlyIf = NativeBuild.class)
-    @Record(value = ExecutionTime.RUNTIME_INIT)
-    void warnJvmInNative(JvmOnlyRecorder recorder) {
-        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
-        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection() {
+        return new ReflectiveClassBuildItem(false, false, new String[] {
+                org.iq80.leveldb.impl.Iq80DBFactory.class.getName(),
+                org.apache.camel.support.DefaultExchangeHolder.class.getName()
+        });
```

Comment:
Style tip: the last arg of `ReflectiveClassBuildItem` constructor is multiarg, so you could make it shorter by omitting the the array declaration (it would compile to exactly the same bytecode):

```suggestion
        return new ReflectiveClassBuildItem(false, false, 
                org.iq80.leveldb.impl.Iq80DBFactory.class.getName(),
                org.apache.camel.support.DefaultExchangeHolder.class.getName());
```

Is org.apache.camel.support.DefaultExchangeHolder really necessary? IIRC, there was some discussion around this recently.

Reply:
done